### PR TITLE
Ensure mapped dimensions have the same size

### PR DESCRIPTION
**Pull request by ulysseB**
This PR adds a list of dimensions mapped together to the function. Pairs of dimensions in `dim_map` fields in the operands must now be mapped. A new constraint ensure mapped dimensions have the same size.

Mapped dimensions should also help us improve the data flow encoding: it exposes the structure of `dim_map`s to the constraint programing framework.

### CODE REVIEW

File: src/ir/function.rs [hunk 1]
```diff
@@ -498,60 +557,62 @@ impl<'a> Function<'a> {
...
         ));
         self.insts.set_lazy(lowered.load, ld);
         self.insts[dst_inst].lower_dim_map(dst_operand_pos, lowered.load, ld_dim_map);
 
         Ok(lowered)
     }
+
+    /// Activate dimensions mapped to existing ones. Takes as input a list of mappings, in the
+    /// format `(mapping id, [old dim, new dim])`. Returns the mapping from old dimensions to new
+    /// ones if `old_to_new` is true. The mapping from new dimensions to old ones otherwise.
```
Comment (nicoTolly) [inline src/ir/function.rs]: Elaborate on this comment : mappings are symmetric, so it's weird that we have to care about the order in which they are returned. This is important  for the functions that take this mapping as input

Comment (nicoTolly) [reply src/ir/function.rs]: Also, the name of the function is a bit weird. Maybe create_mapped_dims ? Or promote_mapped_dims ? Could we find some verb better than "activate" for "this potential dimension that was referenced is now promoted to a concrete dimension and exists in the model" ? Don't really have a great suggestion on that

Comment (ulysseB) [reply src/ir/function.rs]: I don't have a better solution either, so I will just explain in details what is does in the comment

**Pull request merged**